registration_test.go

```diff
@@ -252,21 +252,35 @@ func TestUnregister(t *testing.T) {
 
 func TestInstall(t *testing.T) {
 	if wsl.MockAvailable() {
-		t.Skip("Skipping because the mock does not capture the compexity of this function")
+		t.Parallel()
 	}
 
+	type backend = int
+	const (
+		either backend = iota
+		mockOnly
+		realOnly
+	)
+
 	testCases := map[string]struct {
 		distroName       string
 		appxPackage      string
 		precancelContext bool
 
+		backend backend
+		mockErr bool
+
 		wantErr bool
 	}{
 		"Success with a real distro name": {distroName: "Ubuntu-22.04", appxPackage: "CanonicalGroupLimited.Ubuntu22.04LTS"},
 
-		"Error with a not real distro name": {distroName: "Ubuntu-00.04", wantErr: true},
-		"Error with an empty string":        {distroName: "", wantErr: true},
-		"Error with a cancelled context":    {distroName: "Ubuntu-22.04", precancelContext: true, wantErr: true},
+		// Misuse errors
+		"Error with an empty string":     {distroName: "", wantErr: true},
+		"Error with a cancelled context": {distroName: "Ubuntu-22.04", precancelContext: true, wantErr: true},
+
+		// Backend-specific errors
+		"Error with a not real distro name": {distroName: "Ubuntu-00.04", backend: realOnly, wantErr: true},
```

Comment:
do you think that’s worth having a parameter for this, or just hardcoding distroName with maybe a more obvious name that it does not exists, and have the backend reacting to this?

Reply:
I'd rather keep it consistent with the rest of the mock, i.e. have an extra variable. Otherwise it'd be the only place in the package where we signal the mock via the name of the distro.

Reply:
Yeah, the variable is `InstallError`, it's already there. I'd rather have two tests:
- "Error with a not real distro name": only with a real backend
- "Error due to mock error": only with a mock backend

They essentially exercise the same path (API-wise), but I wanted to keep them separate since they do so in diderent ways and I think that makes the code easier to read.

Reply:
I changed it to `Error from WSL executable...` and `...from executable mock...` to make it clear, as the function itself is called uninstall and that could be confusing 😅.